hathor/metrics.py

```diff
@@ -322,10 +339,45 @@ def set_cache_data(self) -> None:
             if misses:
                 self.transaction_cache_misses = misses
 
+    def set_tx_storage_data(self) -> None:
+        self.log.info("Set tx storage data")
+        store = self.tx_storage
+
+        if isinstance(self.tx_storage, TransactionCacheStorage):
+            store = self.tx_storage.store
+
+        if not isinstance(store, TransactionRocksDBStorage):
+            # We currently only collect metrics for RocksDB
+            return
+
+        try:
+            # Try to use the number of blocks to decide when we should update the tx storage data
+            block_count = self.tx_storage.get_height_best_block()
```

Comment:
Should we use the best block's height or timestamp? If we use the timestamp, we can easily see the evolution of the storage use.

Reply:
I'm not sure I understand what you mean. Why couldn't we easily see the evolution of storage usage by using the blocks height as interval for the updates?

Reply:
But just to explain the reasoning behind using the blocks height, I did this specifically to avoid having a wrong metric value in the first 24h of the fullnode's life.

If using solely the timestamp, we would get the first measure still during sync, and the storage size would still be wrong.

Then we would get the right measure only 24h later.

By using blocks height, I was able to keep updating the storage size in smaller intervals during sync until it is finished, and then we would have approximately the same interval of 24h for updating it again.

Reply:
I got it. It's a clever solution. But we might use the block's timestamp instead of the block's height. Both are increasing sequences but the timestamp would allow us to show a date/time in the x-axis (instead of the block number).

Reply:
Just registering here what we talked externally, actually the x-axis in Prometheus will still contain a timestamp, this is not determined by the use of the block height.

All metrics in Prometheus are time-based and the timestamp assigned to them is the moment Prometheus collects them from its targets.

Reply:
Got it. Thanks for the explanation.